### PR TITLE
Add Github CI

### PR DESCRIPTION
**Pull request by Ferret-san**
Adding basic Github CI workflow to streamline testing and release process

### CODE REVIEW

**Review by jcstein — COMMENTED** [review]
utACK. added @MSevey for review as he is a github guru

File: .github/workflows/cannon.yml [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+            echo '{
+              "name": "contracts-bedrock",
+              "version": "1.0.0",
+              "private": true,
+              "description": "Optimism Bedrock smart contracts",
+              "scripts": {
+                "test": "echo \"Error: no test specified\" && exit 1"
+              }
+            }' > package.json
+          fi
```
Comment (MSevey) [inline .github/workflows/cannon.yml:61]: If these files and packages need to be present, why not just commit them? 
If it is an npm package, there should always be a package.json file. It doesn't really make sure to be creating that in CI.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
+name: CI
```
Comment (MSevey) [inline .github/workflows/ci.yml:1]: nit: I'd name this something more descriptive like `docker`.

File: .github/workflows/validate.yml
Comment (MSevey) [inline .github/workflows/validate.yml:1]: fwiw I've always viewed this check as not that useful in CI. 
It is a great local linter, but by the time you push your code to CI, if your workflow isn't valid, it just fails and tells you the error.

**Review by MSevey — APPROVED** [review]
lgtm, just left some thoughts to consider. nothing blocking though

File: .github/workflows/cannon.yml (hunk 1 above)
Comment (Ferret-san) [reply .github/workflows/cannon.yml:61]: Yeah agree that would be more optimal, can change it later if this causes issues as this is all repurposed from the official OP CI workflow

File: .github/workflows/validate.yml
Comment (Ferret-san) [reply .github/workflows/validate.yml:1]: ok, I can remove it then (it was really annoying to deal with lol)

**Pull request merged**